tests/api_tests/comments/test_views.py

```diff
@@ -0,0 +1,889 @@
+        self.comment.reports = self.comment.reports or {}
+        self.comment.reports[self.user._id] = {'category': 'spam', 'text': 'This is spam'}
+        self.comment.save()
+
+        self.public_comment = CommentFactory(node=self.public_project, target=self.public_project, user=self.contributor)
+        self.public_comment.reports = self.comment.reports or {}
+        self.public_comment.reports[self.user._id] = {'category': 'spam', 'text': 'This is spam'}
+        self.public_comment.save()
+
+        self.private_url = '/{}comments/{}/reports/'.format(API_BASE, self.comment._id)
+        self.public_url = '/{}comments/{}/reports/'.format(API_BASE, self.public_comment._id)
+
+        self.payload = {
+            'data': {
+                'type': 'comment_reports',
+                'attributes': {
+                    'category': 'spam',
+                    'message': 'delicious spam'
+                }
+            }
+        }
+
+    def test_private_node_logged_out_user_cannot_view_reports(self):
+        res = self.app.get(self.private_url, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    def test_private_node_logged_in_non_contributor_cannot_view_reports(self):
+        res = self.app.get(self.private_url, auth=self.non_contributor.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_private_node_only_reporting_user_can_view_reports(self):
+        res = self.app.get(self.private_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        report_json = res.json['data']
+        report_ids = [report['id'] for report in report_json]
+        assert_equal(len(report_json), 1)
+        assert_in(self.user._id, report_ids)
+
+    def test_private_node_reported_user_does_not_see_report(self):
+        res = self.app.get(self.private_url, auth=self.contributor.auth)
+        assert_equal(res.status_code, 200)
+        report_json = res.json['data']
+        report_ids = [report['id'] for report in report_json]
+        assert_equal(len(report_json), 0)
+        assert_not_in(self.contributor._id, report_ids)
+
+    def test_public_node_only_reporting_contributor_can_view_report(self):
+        res = self.app.get(self.public_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        report_json = res.json['data']
+        report_ids = [report['id'] for report in report_json]
+        assert_equal(len(report_json), 1)
+        assert_in(self.user._id, report_ids)
+
+    def test_public_node_reported_user_does_not_see_report(self):
+        res = self.app.get(self.public_url, auth=self.contributor.auth)
+        assert_equal(res.status_code, 200)
+        report_json = res.json['data']
+        report_ids = [report['id'] for report in report_json]
+        assert_equal(len(report_json), 0)
+        assert_not_in(self.contributor._id, report_ids)
+
+    def test_public_node_non_contributor_does_not_see_report(self):
```

Comment:
Per the above discussion about controlling access to comment reports: If someone is not a contributor to the node, then presumably they perform no admin role whatsoever in the life of this project. 

Why do we allow them to access this endpoint at all? Instead of returning a blank body, why not a 403?